### PR TITLE
feat: Implement skeleton PubSub

### PR DESCRIPTION
**Pull request by FelisiaM**
### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [ ] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: terraform/pubsub/bind/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+output "name" { value = google_service_account.account.name }
```
Comment (blgm) [inline terraform/pubsub/bind/outputs.tf:1]: If the aim is to match the legacy broker as closely as possible, would `Name` and `Email` be better?

**Review by blgm — APPROVED** [review]

File: google-pubsub.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+support_url: https://cloud.google.com/support/
+tags: [gcp, pubsub, google-pubsub, beta]
+plans: []
+provision:
+  user_inputs:
+  - field_name: topic_name
+    type: string
+    details: The name of the topic to create
+    default: csb-topic-${request.instance_id}
+  - field_name: subscription_name
```
[the remark this answers is not in the document]
Comment (FelisiaM) [reply google-pubsub.yml:18]: Yeah that is correct. 
The aim of this PR is to create the skeleton of PubSub with the same functionality as legacy offering. I suspect we might think longer about this and maybe offer it slightly differently.

**Review by blgm — COMMENTED** [review]
Impressive to have built this in just one day.

The thing that we are prioritising above all else, is for Java Spring apps that worked with the old GCP broker to "just work" with this service. I'd really like to see a Java Spring test app to test this, and I'd like to see the app verified by running against the old broker. Reading the old broker, I'm not sure whether the binding had [`TopicName` and `Subscription`](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/2f676b434e3f9d570f3b1d831f81e3a3de8c7664/pkg/providers/builtin/pubsub/broker.go#L103-L104) in the binding output, or [`topic_name` and `subscription_name`](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/2f676b434e3f9d570f3b1d831f81e3a3de8c7664/pkg/providers/builtin/pubsub/definition.go#L116-L138).

File: acceptance-tests/pubsub_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			services.WithParameters(map[string]any{"subscription_name": random.Name()}))
+		defer serviceInstance.Delete()
+
+		By("pushing the unstarted app twice")
+		publisherApp := apps.Push(apps.WithApp(apps.PubSubApp))
+		subscriberApp := apps.Push(apps.WithApp(apps.PubSubApp))
+		defer apps.Delete(publisherApp, subscriberApp)
+
+		By("binding the apps to the storage service instance")
+		binding := serviceInstance.BindWithParams(publisherApp, `{"role":"pubsub.editor"}`)
```
Comment (blgm) [inline acceptance-tests/pubsub_test.go:28]: This is the default role. I'd be inclined to leave it out, or do a more interesting test with non-default roles.

File: google-pubsub.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (FelisiaM) [reply google-pubsub.yml:18]: That also reminded me to remove the default and set it to `""`

File: acceptance-tests/pubsub_test.go (hunk 3 above)
Comment (FelisiaM) [reply acceptance-tests/pubsub_test.go:28]: Oh yeah I tried doing one as "publisher" role and one as "subscriber". It did highlight issues that I will look into tomorrow.

**Pull request merged**